sqllogictest/src/runner.rs

```diff
@@ -328,6 +329,130 @@ impl<D: AsyncDB> Runner<D> {
     }
 }
 
+/// A parallel Sqllogictest runner
+///
+/// before beginning the runner, we should use prepare(_async) to parse slt files and split task and
+/// then we can pass the tasks to runner for parallel running.
```

Comment:
Is it possible to merge the prepare and run function into one?
One step further, is it possible not to introduce a new runner?